src/Microsoft.AspNet.Razor.Runtime/TagHelpers/DefaultTagHelperContent.cs

```diff
@@ -38,15 +44,12 @@ public override bool IsWhiteSpace
         {
             get
             {
-                foreach (var value in _buffer)
+                if (_buffer == null)
```

Comment:
@NTaylorMullen huh? `IsModified` muddies the protections here. This really is a `null` check to avoid NREs in `CheckIfBufferedHtmlContentIsWhiteSpace()`.